src/domain/authentication/userManager.ts

```diff
@@ -30,31 +30,20 @@ const settings: UserManagerSettings = {
 
 const userManager = createUserManager(settings);
 
-userManager.events.addUserSessionChanged(() => {
-  console.log('zzz user session changed');
-});
-
-userManager.events.addAccessTokenExpiring(() => {
-  console.log('userManager - addAccessTokenExpiring - fetching new token');
-  // TODO: Find out if this is needed and smart.
-  userManager
-    .getUser()
-    .then((user) => {
-      if (user?.access_token) {
-        console.log(user.access_token);
-        fetchApiToken(user?.access_token)
-          .then((apiToken) => {
-            localStorage.setItem('apiToken', apiToken);
-          })
-          .catch((error) => {
-            console.error('fetchApiToken in expring caught error');
-            console.error(error);
-          });
-      }
-    })
-    .catch((error) => {
-      console.error('Could not find user');
-    });
+userManager.events.addAccessTokenExpiring(async () => {
+  console.count('addAccessTokenExpiring');
+  try {
+    console.count('addAccessTokenExpiring - signinSilent');
+    const newUser = await userManager.signinSilent();
+    await userManager.storeUser(newUser);
+    const apiToken = await fetchApiToken(newUser.access_token);
+    console.count('userManager addAccessTokenExpired fetchApiToken');
+    localStorage.setItem('apiToken', apiToken);
+    console.count('setting fetchingApiToken in localstorage to 0');
```

Comment:
This message seems to be off?

Reply:
I updated the PR with some small improvements. Good spot!